File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.cpp
#include <string>
#include <cmath>

using namespace std;

int solution(int n) {
    int answer = 0;
    string three;
    
    while(n > 0){
        three += to_string(n % 3);
        n /= 3;
    }
    
    int len = three.length() - 1;
    for(int i = len; i >= 0 ; i--){
        answer += (three[i] - '0') * pow(3, len - i);
    }
     
    return answer;
}

File: Python3/프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.py
def solution(n):
    t, answer, cnt = '', 0, 0
    
    while n > 0:
        t += str(n % 3)
        n //= 3
        
    for i in reversed(t):
        answer += int(i) * (3 ** cnt)
        cnt += 1
        
    return answer
        